app/controllers/users_controller.rb

```diff
@@ -23,10 +33,50 @@ def create
     end
   end
 
+  def update
+    if @user.update user_params
+      flash[:success] = t "form.update_success"
+      redirect_to @user
+    else
+      flash.now[:danger] = t "form.update_fail"
+      render :edit
+    end
+  end
+
+  def destroy
+    if @user&.destroy
+      flash[:success] = t "general.delete_success"
+    else
+      flash[:danger] = t "general.delete_fail"
+    end
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params.require(:user).permit(:name, :email, :password,
                                  :password_confirmation)
   end
+
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = t "general.please_login"
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    redirect_to(root_url) unless current_user? @user
+  end
+
+  def admin_user
+    redirect_to(root_url) unless current_user.admin?
+  end
+
+  def find_user
+    @user = User.find_by id: params[:id]
+    redirect_to root_url if @user.nil?
```

Comment:
Em di chuyển sang root thì cũng cho ng ta 1 cái lý do chứ, đâu phải `chia tay là ko cần lý do đâu ` 
```suggestion
    return if @user
    
    flash[:warning] = "User not found# I18n nhé
    redirect_to root_path
```